library/std/arithmetic.qs

```diff
@@ -289,4 +316,167 @@ namespace Microsoft.Quantum.Arithmetic {
         }
     }
 
+    //
+    //
+    //      New arithmetic operations starts here.
+    // Once it is done, previous implementation will be removed.
+    //
+    //
+
+    //
+    // Operation: Add      |    Ripple-carry    | Carry look-ahead
+    // ____________________________________________________________
+    // y += 5              |  RippleCarryIncByL✔|              N/A
+    // y += x              | RippleCarryIncByLE✔| LookAheadIncByLE
+    // z = x + 5 (z was 0) |                N/A |              N/A
+    // z = x + y (z was 0) |   RippleCarryAddLE✔|   LookAheadAddLE
+    // z += x + 5          |                N/A |              N/A
+    // z += x + y          |                N/A |              N/A
+    //
+
+    /// # Summary
+    /// Increment a little-endian ys register by a BigInt number c
+    ///
+    /// # Description
+    /// Compute ys += c modulo 2ⁿ, where ys is a little-endian register
+    /// of length n > 0, c >= 0 is a BigInt constant, and c < 2ⁿ.
+    /// This operation can be used when the choice of implementation
+    /// is not important.
+    @Config(Full)
+    operation IncByL (c : BigInt, ys : Qubit[]) : Unit is Adj + Ctl {
+        RippleCarryIncByL(c, ys);
+    }
+
+    /// # Summary
+    /// Increments a little-endian register ys by a little-endian register xs
+    ///
+    /// # Description
+    /// Compute ys += xs modulo 2ⁿ, where xs and ys are little-endian registers,
+    /// n = Length(ys), and Length(xs) <= Length(ys).
+    /// This operation can be used when the choice of implementation
+    /// is not important.
+    @Config(Full)
+    operation IncByLE (xs : Qubit[], ys : Qubit[]) : Unit is Adj + Ctl {
+        RippleCarryIncByLE(xs, ys);
+    }
+
+    /// # Summary
+    /// Set a zero-initialized little-endian register zs to the sum of
+    /// little-endian registers xs and ys
+    ///
+    /// # Description
+    /// Compute zs := xs + ys, where xs, ys, and zs are little-endian
+    /// registers, Length(xs) = Length(ys), Length(xs) <= Length(zs) <= Length(xs)+1,
+    /// assuming zs is 0-initialized.
+    /// This operation can be used when the choice of implementation
+    /// is not important.
+    @Config(Full)
+    operation AddLE (xs : Qubit[], ys : Qubit[], zs : Qubit[]) : Unit is Adj {
+        RippleCarryAddLE(xs, ys, zs);
+    }
+
+    /// # Summary
+    /// Increment a little-endian ys register by a BigInt number c
+    /// using the ripple-carry algorithm.
+    ///
+    /// # Description
+    /// Compute ys += c modulo 2ⁿ, where ys is a little-endian register
+    /// of length n > 0, c >= 0 is a BigInt constant, and c < 2ⁿ.
+    /// This operation uses the ripple-carry algorithm.
+    @Config(Full)
+    operation RippleCarryIncByL (c : BigInt, ys : Qubit[]) : Unit is Adj + Ctl {
+        let ysLen = Length(ys);
+        Fact(ysLen > 0, "Length of `ys` must be at least 1");
```

Comment:
There is inconsistency in whether diagnostic messages start lower case (see above) or upper case (here).

Reply:
Typically error messages start with capital letter. They start with the small letter if the first word is the reference to one of the arguments. I'll add quotations if it helps.